org.sonarlint.eclipse.m2e/src/org/sonarlint/eclipse/m2e/internal/M2eUtils.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * SonarLint for Eclipse
+ * Copyright (C) 2015-2017 SonarSource SA
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.m2e.internal;
+
+import org.eclipse.core.internal.localstore.FileSystemResourceManager;
+import org.eclipse.core.internal.resources.ICoreConstants;
+import org.eclipse.core.internal.resources.Resource;
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.internal.IMavenConstants;
+import org.eclipse.m2e.core.project.IMavenProjectFacade;
+import org.eclipse.m2e.core.project.IMavenProjectRegistry;
+import org.sonarlint.eclipse.core.SonarLintLogger;
+
+public class M2eUtils {
+
+  private M2eUtils() {
+  }
+
+  /**
+   * m2e creates IProject for every module (flat layout). It means the same file can appear in several projects.
+   * When it is the case we want to keep the most specific one.
+   */
+  public static boolean isInNestedModule(IFile file) {
+    IProject project = file.getProject();
+    try {
+      if (project.hasNature(IMavenConstants.NATURE_ID)) {
+        IMavenProjectRegistry projectManager = MavenPlugin.getMavenProjectRegistry();
+
+        IMavenProjectFacade projectFacade = projectManager.create(project, null);
+        if (projectFacade != null && "pom".equals(projectFacade.getPackaging())) {
+          return !toSpecificFile(file).equals(file);
+        }
+      }
+    } catch (CoreException ex) {
+      SonarLintLogger.get().error(ex.getMessage(), ex);
+    }
+    return false;
+  }
+
+  private static IFile toSpecificFile(IFile file) {
+    IFile finalFile = file;
+    IPath rawLocation = file.getRawLocation();
+    if (rawLocation != null) {
+      // TODO use getFileForLocation when we will support only Eclipse 4.6+
+      // IFile moreSpecific = ResourcesPlugin.getWorkspace().getRoot().getFileForLocation(rawLocation);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)